test/dotcom/alerts/subway_test.exs

```diff
@@ -0,0 +1,229 @@
+      # Exercise/Verify
+      assert group_order() |> is_list()
+    end
+  end
+
+  describe "groups/0" do
+    test "returns an ordered list" do
+      # Setup
+      group_order = group_order()
+
+      # Exercise
+      groups = groups()
+
+      # Verify
+      assert groups |> List.first() |> Kernel.elem(0) == Enum.at(group_order, 0)
+      assert groups |> List.last() |> Kernel.elem(0) == Enum.at(group_order, -1)
+    end
+  end
+
+  describe "find_group/1" do
+    test "returns 'Other' when the group is not found" do
+      # Setup
+      effect = :foobarbaz
+
+      # Exercise
+      group = find_group(effect)
+
+      # Verify
+      assert group == "Other"
+    end
+
+    test "returns the correct group for grouped effects" do
+      # Setup
+      groups = groups() |> Enum.reject(fn {group, _effects} -> group == "Other" end)
+      random_group = Enum.random(groups) |> Kernel.elem(0)
+
+      effects =
+        Enum.find(groups, fn {group, _effects} -> group == random_group end) |> Kernel.elem(1)
+
+      random_effect = Enum.random(effects)
+
+      # Exercise
+      group = find_group(random_effect)
+
+      # Verify
+      assert group == random_group
+    end
+  end
+
+  describe "sort_alerts/1" do
+    test "alerts are sorted by station, and then start time" do
+      # Setup
+      earlier = random_date_time()
+      later = random_time_range_date_time({earlier, nil})
+
+      a_station = Factories.Stops.Stop.build(:stop, station?: true, name: "A")
+      b_station = Factories.Stops.Stop.build(:stop, station?: true, name: "B")
+
+      stub(Stops.Repo.Mock, :get, fn id ->
+        case id do
+          "A" -> a_station
+          "B" -> b_station
+        end
+      end)
+
+      a_stop = MapSet.new(["A"])
+      b_stop = MapSet.new(["B"])
+
+      a_informed_entity =
+        Factories.Alerts.InformedEntitySet.build(:informed_entity_set, stop: a_stop)
+
+      b_informed_entity =
+        Factories.Alerts.InformedEntitySet.build(:informed_entity_set, stop: b_stop)
+
+      # 'A' is first because of its station and start time.
+      a_alert =
+        Factories.Alerts.Alert.build(:alert,
+          id: "A",
+          active_period: [{earlier, nil}],
+          informed_entity: a_informed_entity
+        )
+
+      # 'B' is second because of its station and start time.
+      b_alert =
+        Factories.Alerts.Alert.build(:alert,
+          id: "B",
+          active_period: [{later, nil}],
+          informed_entity: a_informed_entity
+        )
+
+      # 'C' is last even though its active period is first. Its station is last.
+      c_alert =
+        Factories.Alerts.Alert.build(:alert,
+          id: "C",
+          active_period: [{earlier, nil}],
+          informed_entity: b_informed_entity
+        )
+
+      alerts = [c_alert, b_alert, a_alert]
+
+      # Exercise
+      sorted_alerts = sort_alerts(alerts)
+
+      # Verify
+      assert [a_alert, b_alert, c_alert] == sorted_alerts
+    end
+  end
```

Comment:
`"A"`, `"B"`, and `"C"` can be replaced with faker values. Probably with `Faker.Util.sample_uniq` in order to ensure that the values are unique.